blockdevice/util_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blockdevice
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestGetCapacity(t *testing.T) {
+	var tests = map[string]struct {
+		src   unstructured.Unstructured
+		isErr bool
+	}{
+		"kind mismatch error": {
+			src: unstructured.Unstructured{
```

Comment:
Have checks for nil & empty unstruct instances

Reply:
added